### DISCUSSION TITLE
Completely removing a command from the internal database

### DISCUSSION THREAD

**Discussion by iiPythonx** [Rapptz/discord.py#5997] [Q&A] [answered]
So, I'm probably doing this the wrong way and if so then please tell me.
Rather than having a core for my bot that depends on being imported, I decided "why not just pass the core to all the commands when the bot loads them"? So, I did some digging and decided that I wasn't able to use `bot.load_extension` since it doesn't allow you to pass data to the `setup` function in cogs.

Alas, I'm using `importlib` to manually import the modules and them load them using `bot.add_cog(Command(bot, core))`.
However, the problem is that I'm trying to make a reload command and I have absolutely no idea how to remove a command.

Normally I would use `bot.unload_extension()` but it just returns `Module not loaded` (understandably).
I've tried `bot.commands.discard()`, `bot.remove_command()`, etc. Some of them don't unload at all, others actually work and stop allowing me to use the command.

However, on all of them, the result is the same, the command isn't reloaded (it's the same as it originally was).

Any ideas? If theres a better way of doing what I'm trying to do then please let me know.

**Comment by Rapptz** [comment, marked as answer]
Rather than re-inventing the extension machinery you can just make `core` an attribute of `bot` and access it via `bot.core` in the `setup` function.

**Reply by iiPythonx** [reply]
This is why I post questions, makes a lot better sense now thanks! :)
